frontend/src/pages/Auth/AdminForm.tsx

```diff
@@ -0,0 +1,214 @@
+import { useAuthContext } from '@authentication/AuthContext'
+import {
+	AutoJoinEmailsInput,
+	getEmailDomain,
+} from '@components/AutoJoinEmailsInput'
+import { Button } from '@components/Button'
+import {
+	AppLoadingState,
+	useAppLoadingContext,
+} from '@context/AppLoadingContext'
+import { useUpdateAdminAndCreateWorkspaceMutation } from '@graph/hooks'
+import {
+	Box,
+	ButtonLink,
+	Callout,
+	Form,
+	Stack,
+	Text,
+	useFormState,
+} from '@highlight-run/ui'
+import { AuthBody, AuthFooter, AuthHeader } from '@pages/Auth/Layout'
+import { Landing } from '@pages/Landing/Landing'
+import analytics from '@util/analytics'
+import { getAttributionData } from '@util/attribution'
+import { message } from 'antd'
+import React, { useEffect, useState } from 'react'
+import { useNavigate } from 'react-router-dom'
+
+import * as styles from './AdminForm.css'
+import * as authRouterStyles from './AuthRouter.css'
+
+const COMMON_EMAIL_PROVIDERS = ['gmail', 'yahoo', 'hotmail', 'fastmail', 'protonmail', 'hey.com']
+
+export const AdminForm: React.FC = () => {
+	const [showPromoCodeField, setShowPromoCodeField] = useState(false)
+	const { setLoadingState } = useAppLoadingContext()
+	const { admin, refetchAdmin } = useAuthContext()
+	const navigate = useNavigate()
+	const [updateAdminAndCreateWorkspace, { loading }] =
+		useUpdateAdminAndCreateWorkspaceMutation()
+
+	if (admin?.about_you_details_filled) {
+		navigate('/setup')
+	}
+
+	const adminEmailDomain = getEmailDomain(admin?.email)
+	const isCommonEmailDomain = COMMON_EMAIL_PROVIDERS.some(
+		(p) => adminEmailDomain.indexOf(p) !== -1,
+	)
+
+	const formState = useFormState({
+		defaultValues: {
+			firstName: '',
+			lastName: '',
+			role: '',
+			companyName: '',
+			autoJoinDomains: '',
+			promoCode: '',
+		},
+	})
+
+	formState.useSubmit(async () => {
+		if (!formState.valid) {
+			analytics.track('About you submission failed')
+			formState.setError(
+				'__error',
+				'Please fill out all form fields correctly.',
+			)
+			return
+		}
+
+		analytics.track('About you submitted')
+
+		try {
+			const attributionData = getAttributionData()
+			await updateAdminAndCreateWorkspace({
+				variables: {
+					admin_and_workspace_details: {
+						first_name: formState.values.firstName,
+						last_name: formState.values.lastName,
+						user_defined_role: formState.values.role,
+						workspace_name: formState.values.companyName,
+						allowed_auto_join_email_origins:
+							formState.values.autoJoinDomains,
+						promo_code: formState.values.promoCode || undefined,
+						...attributionData,
+					},
+				},
+			})
+
+			message.success(
+				`Nice to meet you ${formState.values.firstName}, let's get started!`,
```

Comment:
Can we put that inside onComplete of `updateAdminAndCreateWorkspace` and handle errors inside onError to avoid `catch (e:any)` ?

Reply:
I looked at this, but wouldn't we still end up with an `any` type on the error?

<img width="257" alt="image" src="https://user-images.githubusercontent.com/308182/219802896-d3f54027-beef-4ee3-8891-c8f20db4366f.png">